### PR TITLE
Add simple_efficiency tests for numpy array inputs

### LINKED ISSUE #2649: Add simple_efficiency tests to cover numpy arrays and pandas Series
### Description

`pvlib.transformer.simple_efficiency` supports vectorized operations via
NumPy, but the current test coverage primarily exercises scalar inputs.

This issue proposes adding tests that explicitly cover
`numpy.ndarray` inputs for:

- `input_power`
- `load_loss`
- `no_load_loss`

including the edge case `load_loss = 0`.

### PR DESCRIPTION
**Pull request by aman-coder03**
this PR adds test coverage for vectorized use of `pvlib.transformer.simple_efficiency` with `numpy.ndarray` inputs.

the tests verify:
- support for NumPy array inputs for `input_power`, `load_loss`,
  and `no_load_loss`
- consistency between vectorized and scalar evaluations

No functional changes are introduced.

fixes #2649

### CODE REVIEW

**Comment by aman-coder03** [comment]
the CI failures are due to deprecation-enforcement tests in `tests/test_irradiance.py`, unrelated to this PR.
this PR only modifies `tests/test_transformer.py`.
please let me know if you would like me to rebase

**Comment by cwhanse** [comment]
@aman-coder03 this is the scope of #2658 

Could you complete #2646 first? It needs a whatsnew note.

**Comment by aman-coder03** [comment]
@cwhanse shall I create `pvlib-python\docs\sphinx\source\whatsnew\v0.13.2.rst` ?
and then add 
```
* Fix a division-by-zero condition in
  :py:func:`~pvlib.transformer.simple_efficiency` when ``load_loss = 0``.
  (:issue:`2645`)
```
in Bug Fixes

**Comment by cwhanse** [comment]
It should be v0.14.1.rst. Thanks!

**Comment by aman-coder03** [comment]
@cwhanse i have completed that, you can have a look!!

**Comment by aman-coder03** [comment]
@cwhanse as suggested by you in #2646 , i have implemented the additional tests in this separate PR to avoid scope creep in #2646. Let me know if you would like any adjustments.

File: tests/test_transformer.py [hunk 1]
```diff
@@ -58,3 +60,41 @@ def test_simple_efficiency_known_values():
                                       *args),
         rating,
     )
+
+
+def test_simple_efficiency_numpy_array_inputs():
```
Comment (cwhanse) [inline tests/test_transformer.py]: @aman-coder03 I think it is premature to test with all inputs as np.array. In other discussions we don't have consensus to allow the parameters `no_load_loss` and `load_loss` to be anything other than `float`, and intend to edit the docstring accordingly.

If we agree in the future to allow vectors for these parameters, then this test would be needed.

I don't want to include it now so to prevent someone's AI from deciding we have missed a combination of types in other tests.

Comment (aman-coder03) [reply tests/test_transformer.py]: Would you prefer that I remove the test that uses array inputs for `no_load_loss` and `load_loss`, and keep only the vectorized `input_power` case? Or should I drop the vectorized tests entirely for now?

Comment (cwhanse) [reply tests/test_transformer.py]: I think drop `test_simple_efficiency_numpy_array_inputs` and keep `test_simple_efficiency_vector_equals_scalar`

Comment (aman-coder03) [reply tests/test_transformer.py]: implemented the suggested changes @cwhanse , thanks for your guidance!!

**Comment by cwhanse** [comment]
Looks OK now. Please add a brief note in the Testing section of this file: https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/whatsnew/v0.15.1.rst

**Comment by aman-coder03** [comment]
added whatsnew entry in the testing section, please have a look.
thanks again!! @cwhanse

**Comment by echedey-ls** [comment]
I suggest to merge the tests into one, most functions are type-tested in the same test for any numeric implementation, and it can be shortened with the `@pytest.mark.parametrize`.

File: docs/sphinx/source/whatsnew/v0.15.1.rst [hunk 2]
```diff
@@ -28,7 +28,9 @@ Documentation
 
 Testing
 ~~~~~~~
-
+* Add test to verify that :py:func:`~pvlib.transformer.simple_efficiency`
+  produces consistent results for vectorized and scalar inputs.
+  (:pull:`2661`)
```
Comment (echedey-ls) [inline docs/sphinx/source/whatsnew/v0.15.1.rst]:
```suggestion
  (:issue:`2649`, :pull:`2661`)
```

**Review by echedey-ls — COMMENTED** [review]
> I suggest to merge the tests into one, most functions are type-tested in the same test for any numeric implementation, and it can be shortened with the `@pytest.mark.parametrize`.

@aman-coder03 , for my previous comment, I suggest you give it a try doing it without AI, it's not difficult to understand, you have an example in the same file, documentation is public and easily accessible, and we prefer contributors to know what they are doing rather going back and forth to an LLM model.

**Comment by aman-coder03** [comment]
I refactored the numeric tests into a single parametrized test following the pattern used elsewhere in the file. I reviewed the `pytest` documentation  to make sure I understood  the approach
Please let me know if this structure looks good @echedey-ls thanks!!

**Comment by aman-coder03** [comment]
@echedey-ls please have a look at the updated tests

**Comment by echedey-ls** [comment]
I know you are waiting for reviews. Don't ping me every day, it's been a ping per day last three days. I will review whenever I can.

**Comment by kandersolar** [comment]
@aman-coder03 please understand that pvlib's maintainers (like most open-source maintainers) cannot work full-time on pvlib.  We have many other responsibilities and have to fit in time to work on pvlib when we can, and there are always many pvlib things "waiting" for us to have time for them.  In particular we have had many PRs submitted recently that need a lot of work from us as reviewers, putting more demand on our limited time.  

So, a heads up: going forward, we will close PRs that we think will require too much work from reviewers.  In general, pvlib is a difficult project for non-solar energy people to contribute meaningfully to, and we cannot prioritize helping out when we already have many other things to do.

**Comment by aman-coder03** [comment]
Thanks for the clarification and for taking the time to review and merge this. I really appreciate the work you all do to maintain pvlib. i will be more mindful about follow ups in the future
